File: 2015/day_2/part_one.py
def calculate_wrapping_paper(length, width, height):
	top_area = length * width
	front_area = length * height
	side_area = width * height
	return 2 * (top_area + front_area + side_area) + min(top_area, front_area, side_area)

def parse_input(line):
	return [int(num) for num in line.split('x')]


total_wrapping_paper = 0

with open('./input') as f:
	dimensions = f.readlines()
	for dimension in dimensions:
		length, width, height = parse_input(dimension.strip())
		total_wrapping_paper += calculate_wrapping_paper(length, width, height)
print(total_wrapping_paper)	

File: 2015/day_2/part_one.rs
use std::{io::{BufReader, BufRead}, fs::File, str::FromStr, cmp};

fn calculate_wrapping_paper(length: u64, width: u64, height: u64) -> u64 {
  let top_area = length * width;
  let front_area = length * height;
  let side_area = width * height;
  2 * (top_area + front_area + side_area) + cmp::min(top_area, cmp::min(front_area, side_area))
}

fn parse_input(line: String) -> Vec<u64> {
  line.split('x').map(|num| u64::from_str(num).unwrap()).collect()
}

fn main() {
  let file = File::open("./input").unwrap();
  let file_reader = BufReader::new(file);
  let mut required_wrapping_paper: u64 = 0;
  for line in file_reader.lines() {
    let parsed_input = parse_input(line.unwrap());
    required_wrapping_paper += calculate_wrapping_paper(parsed_input[0], parsed_input[1], parsed_input[2]);
  }

  println!("{}", required_wrapping_paper);
}
